src/index.js

```diff
@@ -1,100 +1,93 @@
-import { join, dirname, extname, basename } from 'path'
-import { parse } from 'babylon';
+import { join, dirname, extname, basename } from 'path';
+import loaderUtils from 'loader-utils';
 
 import traverse from './traverse';
+import VirtualModulePlugin from './VirtualModulePlugin';
 
-function parseSource(src) {
-  return parse(src, {
-    sourceType: 'module',
-    plugins: [
-      'asyncFunctions',
-      'jsx',
-      'flow',
-      'classConstructorCall',
-      'doExpressions',
-      'trailingFunctionCommas',
-      'objectRestSpread',
-      'decorators',
-      'classProperties',
-      'exportExtensions',
-      'exponentiationOperator',
-      'asyncGenerators',
-      'functionBind',
-      'functionSent'
-    ]
-  })
-}
-
-function collectStyles(src) {
-  let styles = [];
+function collectStyles(src, tagName = 'css') {
+  const styles = [];
 
   // quick regex as an optimization to avoid parsing each file
-  if (!src.match(/css`([\s\S]*?)`/gmi)) {
+  if (!src.match(new RegExp(`${tagName}\`([\\s\\S]*?)\``, 'gmi'))) {
     return styles;
   }
 
-  let ast = parseSource(src);
-  traverse(ast, {
+  traverse(src, {
     TaggedTemplateExpression(path) {
-      let node = path.node;
+      const node = path.node;
 
-      if (node.tag.name !== 'css' || !path.scope.hasGlobal('css'))
-        return
+      if (node.tag.name !== tagName || !path.scope.hasGlobal(tagName)) {
+        return;
+      }
 
-      let parseError = path.buildCodeFrameError(
+      const parseError = path.buildCodeFrameError(
         'Could not evaluate css. inline css must be statically analyzable'
       );
 
-      let { start, end } = node
+      const { start, end } = node;
 
       // remove the tag and evaluate as a plain template;
-      path.replaceWith(node.quasi)
+      path.replaceWith(node.quasi);
 
-      let { confident, value } = path.evaluate()
+      const { confident, value } = path.evaluate();
 
       if (!confident) {
-        throw parseError
+        throw parseError;
       }
 
-      styles.push({ value, start, end })
-    }
-  })
+      styles.push({ value, start, end });
+    },
+  });
 
-  return styles
+  return styles;
 }
 
 function replaceStyleTemplates(src, styles) {
   let offset = 0;
 
   function splice(str, start, end, replace) {
-    let result = str.slice(0, start + offset) + replace + str.slice(end + offset);
-    offset += replace.length - (end - start)
+    const result = (
+      str.slice(0, start + offset) + replace + str.slice(end + offset)
+    );
+
+    offset += replace.length - (end - start);
     return result;
   }
 
   styles.forEach(({ start, end, path }) => {
-    src = splice(src, start, end, `require('${path}')`)
-  })
+    src = splice(src, start, end, `require('${path}')`);
+  });
 
   return src;
 }
 
+const LOADER_PLUGIN = Symbol('loader added VM plugin');
```

Comment:
Yeah... just wondering if that symbol is needed elsewhere – like if you factor out the virtual module thing and other loaders might use it? idk